docs/appendices/release-notes/unreleased.rst

```diff
@@ -293,3 +293,9 @@ Fixes
     +----+-------------+
     |  1 | {"x": null} |
     +----+-------------+
+
+- Fixed an issue which caused ``INSERT INTO`` statements to skip generated
+  expression validation for partitioned columns.
+
+- Fixed an issue which caused ``INSERT INTO`` statements to leave behind
+  empty partitions if NULL or CHECK constraint on partitioned by column failed.
```

Comment:
Not sure whether we need 2 entries but it looks to me like a separate issues:

1. Validation of gen expression for partitioned by columns was not done at all. https://github.com/crate/crate/issues/14304. we cannot say it left some partitions behind

2. not null check was done on 5.3 but partitions were left behind

Fix tackles both because it about 
do all validation and do it early.

"all" covers gen expressions, others checks are done
"early" covers abandoned partitions